VideoLocker/src/main/java/org/edx/mobile/http/ErrorHandlingCallback.java

```diff
@@ -0,0 +1,266 @@
+package org.edx.mobile.http;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.edx.mobile.util.images.ErrorUtils;
+import org.edx.mobile.view.common.MessageType;
+import org.edx.mobile.view.common.TaskMessageCallback;
+import org.edx.mobile.view.common.TaskProgressCallback;
+
+import java.io.IOException;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import roboguice.RoboGuice;
+
+/**
+ * Generic abstract implementation of Retrofit's
+ * {@link retrofit2.Callback} interface, that takes care of delivering
+ * status and error information to the proper callbacks. It also
+ * provides (and delegates to) a simpler callback interface for
+ * subclasses, stripping out unnecessary parameters, and redirecting
+ * all responses with error codes to the failure callback method (like
+ * it used to be in the implementation in Retrofit 1).
+ *
+ * @param <T> The successful response body type.
+ */
+public abstract class ErrorHandlingCallback<T> implements Callback<T> {
```

Comment:
I think it's fine, and any ambiguity should be obviated by the context in which it would be used; and as I noted before, Retrofit strips the extra contextual information from the classes in favor of brevity.

"server response" would be a totally redundant prefix, since that is exactly the kind of information that's obvious in context. It would also be incomplete, since it doesn't describe the function of the base class at all, and would be no different than naming it just `Callback`, which is the name of the interface that it's implementing.